pkg/kube/cluster-init.sh

```diff
@@ -126,16 +183,39 @@ setup_prereqs () {
         modprobe iscsi_tcp
         #Needed for iscsi tools
         mkdir -p /run/lock
+        mkdir -p "$K3S_LOG_DIR"
         /usr/sbin/iscsid start
         mount --make-rshared /
         setup_cgroup
         #Check network and default routes are up
         wait_for_default_route
         check_network_connection
+        wait_for_device_name
+        chmod o+rw /dev/null
         wait_for_vault
         mount_etcd_vol
 }
 
+config_cluster_roles() {
+        # generate user debugging-user certificates
+        openssl genrsa -out /tmp/user.key 2048
```

Comment:
What are the requirements for this key and certificate? Do they have to be RSA or would ECC work? I assume the cert is self signed. Is that correct?

I want to avoid bringing openssl back in the product and I already have an ECC cert generating golang app.

Reply:
@eriknordmark Yes signed off the k3s server cert.  Where is your ecc cert golang app?  I can look at switching over to that and removing the openssl usage here.

Reply:
I sent you the golang source code for generating certs.

Reply:
latest commit adds cert-gen go tool used to generate certs here.